src/engine/move_handling.h

```diff
@@ -60,18 +60,19 @@ constexpr void performCastleMove(BitBoard& board, const movegen::Move& move, uin
     }
 }
 
+template<Player player>
 constexpr void performPromotionMove(BitBoard& board, const movegen::Move& move, uint64_t& hash)
 {
-    const bool isWhite = board.player == PlayerWhite;
-    const auto type = isWhite ? WhitePawn : BlackPawn;
+    constexpr bool isWhite = player == PlayerWhite;
+    constexpr auto type = isWhite ? WhitePawn : BlackPawn;
 
     // first clear to be promoted pawn
     uint64_t& pawns = isWhite ? board.pieces[WhitePawn] : board.pieces[BlackPawn];
```

Comment:
No need for the tenary. You can just use type.
You can also constexpr the types below. Can't comment/suggest code from the phone :/